pynest/nest/lib/hl_api_connections.py

```diff
@@ -200,69 +202,55 @@ def Connect(pre, post, conn_spec=None, syn_spec=None,
     ---------
     :ref:`connection_mgnt`
     """
-
-    connect_np_arrays = False
-    if isinstance(pre, numpy.ndarray) or isinstance(post, numpy.ndarray):
-        if not (isinstance(pre, numpy.ndarray) and isinstance(post, numpy.ndarray)):
-            raise TypeError("Sources and targets must either both be NodeCollections, "
-                            "or NumPy arrays with conn_spec=None")
-        elif conn_spec is not None:
-            raise ValueError("When connecting two arrays of node IDs, conn_spec cannot be given")
-        elif not (pre.ndim == 1 and post.ndim == 1):
-            raise ValueError("Sources and targets must be 1-dimensional NumPy arrays")
-        else:
-            connect_np_arrays = True
-            conn_spec = 'one_to_one'
+    data_connect, pre, post = _check_input_nodes(pre, post, conn_spec)
```

Comment:
I think the variable name `data_connect` should be changed to better convey that it is a switch to use `connect_arrays()` instead of the normal SLI functions. Perhaps something like `use_connect_arrays` or `connect_arrays_needed`?